app/controllers/animals_controller.rb

```diff
@@ -0,0 +1,53 @@
+class AnimalsController < ApplicationController
+
+  # Show All
+  def index 
+    animals = Animal.all
+    render json: animals
+  end
+
+  # Show One By ID
+  def show
+    animal = Animal.find(params[:id])
+    render json: animal
+  end
+
+  # Create New Animal
+  def create 
+    animal = Animal.create(animal_params)
+    if animal.valid?
+      render json: animal
+    else
+      render json: student.errors 
+    end
+  end
+
+  # Update One Animal By ID
+  def update 
+    animal = Animal.find(params[:id])
+    animal.update(animal_params)
+    if animal.valid?
+      render json: animal
+    else
+      render json: animal.errors 
+    end
+  end
+
```

Comment:
Perfect- I like how you labeled each one of your methods with comments as well. This successfully will allow a user to update an animal in the database